packages/lodestar/src/executionEngine/http.ts

```diff
@@ -261,13 +285,19 @@ export class ExecutionEngineHttp implements IExecutionEngine {
    */
   async getPayload(payloadId: PayloadId): Promise<bellatrix.ExecutionPayload> {
     const method = "engine_getPayloadV1";
-    const executionPayloadRpc = await this.rpc.fetch<
+    const executionPayloadRpc = await this.rpc.fetchWithRetries<
       EngineApiRpcReturnTypes[typeof method],
       EngineApiRpcParamTypes[typeof method]
-    >({
-      method,
-      params: [payloadId],
-    });
+    >(
+      {
+        method,
+        params: [payloadId],
+      },
+      {
+        retryAttempts: this.retryAttempts,
+        retryDelay: this.retryDelay,
+      }
+    );
```

Comment:
Here it would be very important to track retry behaviour. Meaning adding metrics for:
- histogram of retry attempts per call
- histogram of overall requests times to EL

So to de-duplicate code you can add a private method `fetchEl` that handles metrics and calls `fetchWithRetries`. You can even move the logic in `fetchWithRetries` to here since no-one else is using it.

Reply:
bump

Reply:
> You can even move the logic in fetchWithRetries to here since no-one else is using it.

I decided to leave `fetchWithRetries` as is, because as mentioned by @g11tech [here](https://github.com/ChainSafe/lodestar/pull/3854#pullrequestreview-908101788) it makes it easier to test the retry mechanism separately.

@dapplion can you help clarify what you meant by `histogram of retry attempts per call`? I interpreted that to mean another histogram that captures the duration for each retry, while @g11tech interpreted that to mean a counter to try each request. 

Also I was wondering, this suggestion only focuses on adding metrics to retries to done in `executionEngine/http`, while there are other places the `JsonRpcHttpClient` is used, for example in `eth1Provider.ts`. Would it be an idea to move metrics tracking totally into `JsonRpcHttpClient`?  Then every client who uses  `JsonRpcHttpClient` do not need to have metrics separate, it would be generated automatically from using `JsonRpcHttpClient` to make requests.

If yes, then I can do that as another PR. Let me know what you think

Reply:
Actually might be good idea to add metrics to JsonRpcHttpClient. eth1 calls to not the execution engine should have metrics too

Reply:
Actually might be good idea to add metrics to JsonRpcHttpClient. eth1 calls to not the execution engine should have metrics too